textbook/2_grouping.qmd

```diff
@@ -0,0 +1,328 @@
+
+:::
+
+::::
+
+:::::
+
+Coloring data points is a powerful example of a technique known as **grouping**. The goal of grouping is to distinguish between groups by basing the visual appearance of data points on the value of categorical variable. 
+
+As we have seen with alcohol consumption and life expectancy, grouping has major benefits. It allows us to:
+
+1. **Directly compare** trends and values for distinct groups
+
+2. **Discover trends** in the data that were previously obscured
+
+3. **Avoid making inaccurate conclusions** about the relationships in our data
+
+To understand these benefits, let's consider Simpson's Paradox. The paradox was not invented by Homer Simpson et al., but rather Edward H. Simpson, a British codebreaker. 
+
+Simpson reported a curious phenomenon where a trend present in several different groups of data disappears when they are combined. 
+
+Below is an illustration of the Simpson's Paradox at play. Notice how the overall dataset does not have a clear direction but when you color by group, both groups exhibit the same negative trend. 
+
+::: {.panel-tabset}
+
+### Overall
+
+```{python}
+simpsons_df = pd.read_csv("data/simpsons_paradox_data_l.csv")
+alt.Chart(simpsons_df).mark_circle().encode(
+  x=alt.X('X'),
+  y=alt.Y('Y'),
+)
+```
+
+### By group
+
+```{python}
+alt.Chart(simpsons_df).mark_circle().encode(
+  color='Group',
+  x=alt.X('X'),
+  y=alt.Y('Y'),
+)
+```
+
+:::
+
+In this case, grouping makes the insidious Simpson's Paradox visually unmistakable. This is a prime example of the dual purpose of data visualization. It doesn't just present our data to others; it deepens our understanding of it. 
+
+## Strategies for grouping
+
+The grammar of graphics gives us a convenient syntax for grouping. We simply define the encoding channel that should represent a categorical variable, using `aes()` in ggplot and `.encode()` in Altair. Color, shape, size and position are the most effective channels for grouping, and they can be easily interchanged:
+
+::: {.panel-tabset}
+
+# Altair
+::: {.panel-tabset}
+
+# Color
+
+```python
+.encode(color = 'category_column')
+```
+
+# Shape
+
+```python
+.encode(shape = 'category_column')
+```
+
+# Size
+
+```python
+.encode(size = 'category_column')
+```
+
+# Position
+
+```python
+.encode(x = 'category_column')
+```
+
+```python
+.encode(y = 'category_column')
+```
+:::
+
+# ggplot
+::: {.panel-tabset}
+# Color
+
+```python
+aes(color = category_column)
+```
+
+# Shape
+
+```python
+aes(shape = category_column)
+```
+
+# Size
+
+```python
+aes(size = category_column)
+```
+
+# Position
+
+```python
+aes(x = category_column)
+```
+
+```python
+aes(y = category_column)
+```
+:::
+:::
+
+The main challenge in applying grouping is understanding which encoding channel to use to facilitate the comparisons we want to make. Each grouping strategy has unique strengths and pitfalls, which we shall discuss in detail.
+
+### Color
```

Comment:
For some reason, this doesn't quite  match what I see in the rendered version. Maybe the render didn't update after you made another commit? 

![image](https://github.com/user-attachments/assets/9f90eee0-e9ad-485b-9cbd-a405dfa7ee18)